attestation-service/src/verifier/cca/mod.rs

```diff
@@ -77,16 +81,40 @@ impl Verifier for CCA {
             .with_new_session_url(api_endpoint)
             .build()?;
 
-        let nonce = Nonce::Size(32);
+        let mut hasher = Sha384::new();
+        hasher.update(&nonce);
+        hasher.update(&attestation.tee_pubkey.k_mod);
+        hasher.update(&attestation.tee_pubkey.k_exp);
+        let mut hash_of_nonce_pubkey = hasher.finalize().to_vec();
+        hash_of_nonce_pubkey.resize(64, 0);
+
+        log::info!(
+            "HASH(nonce||pubkey):\n\t{}\n",
+            hex::encode(&hash_of_nonce_pubkey)
+        );
+
         let token = evidence.token;
-        match cr.run(nonce, my_evidence_builder, token.clone()).await {
+        let n = Nonce::Value(hash_of_nonce_pubkey.clone());
+        let result = match cr.run(n, my_evidence_builder, token.clone()).await {
             Err(e) => {
                 log::error!("Error: {}", e);
-                return Err(anyhow!("Attestation failed with error: {:?}", e));
-            }
-            Ok(attestation_result) => {
-                log::info!("Attestation Result: {}", attestation_result)
+                bail!("CCA Attestation failed with error: {:?}", e);
             }
+            Ok(attestation_result) => attestation_result,
+        };
+
+        // Get back the pub key to decrypt the ear which holds raw evidence and the session nonce
+        let public_key_pem = verification_api.ear_verification_key_as_pem()?;
+        let dk = jwt::DecodingKey::from_ec_pem(public_key_pem.as_bytes()).unwrap();
+        let plain_ear = Ear::from_jwt(result.as_str(), jwt::Algorithm::ES256, &dk).unwrap();
+
+        let ear_nonce = plain_ear.nonce.unwrap();
```

Comment:
Please try to replace `.unwrap()`s with error handling, s.t. `.context("...")?`

Reply:
Done